indico/web/forms/fields/principals.py

```diff
@@ -144,8 +151,24 @@ def event(self):
         return self.get_form().event
 
     @property
-    def roles(self):
-        return [serialize_role(role) for role in sorted(self.event.roles, key=attrgetter('code'))]
+    def category(self):
+        if self.object_type == 'category':
+            return self.get_form().category
+        return self.event.category
+
+    @property
+    def event_roles(self):
+        return [serialize_event_role(role) for role in sorted(self.event.roles, key=attrgetter('code'))]
+
+    @property
+    def category_roles(self):
+        from indico.modules.categories.util import serialize_category_role
+        category_roles = list(
```

Comment:
see my other comment about the nicer way of getting the role (you could even move the sorting in the query)